internal/exportimport/database/exportimport.go

```diff
@@ -369,6 +369,19 @@ func (db *ExportImportDB) CreateFiles(ctx context.Context, ei *model.ExportImpor
 			logger.Debugw("scheduled new export file for importing", "exportImportID", ei.ID, "filename", fname)
 			insertedFiles++
 		}
+
+		// Close any files that aren't in the list that are retrying.
+		_, err = tx.Exec(ctx, `
+				UPDATE
+					ImportFile
+				SET
+					status = $1
+				WHERE export_import_id=$2 AND NOT zip_filename = ANY($3) AND status = $4
```

Comment:
I think this should just be AND status != complete

if the file was never tried and no longer appears in the index (however unlikely that is) it should still move to a failed state